boring/src/ssl/connector.rs

```diff
@@ -324,8 +328,12 @@ impl SslAcceptor {
     where
         S: Read + Write,
     {
-        let ssl = Ssl::new(&self.0)?;
-        ssl.accept(stream)
+        self.new_session()?.accept(stream)
+    }
+
+    /// Creates a new TLS session, ready to accept a stream.
+    pub fn new_session(&self) -> Result<Ssl, ErrorStack> {
```

Comment:
There is no way to set `TASK_CONTEXT_INDEX` from `tokio_boring::accept` when calling `SslAcceptor::accept`, and given the `Ssl` value doesn't even exist yet I just can't use `SslAcceptor::accept` anymore in `tokio_boring::accept`.